sdk/core/azure-core/src/main/java/com/azure/core/util/ETag.java

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.util.logging.ClientLogger;
+
+/**
+ * This class represents an HTTP ETag. A ETag value could be strong or weak ETag.
+ * More information, check https://en.wikipedia.org/wiki/HTTP_ETag
+ */
+public final class ETag {
+    private static final ClientLogger LOGGER = new ClientLogger(ETag.class);
+
+    private static final String QUOTE_STRING = "\"";
+    private static final String WEAK_ETAG_PREFIX_QUOTE = "W/\"";
+    private static final String ASTERISK = "*";
+
+    /**
+     * The asterisk is a special value representing any resource.
+     */
+    public static final ETag ALL = new ETag(ASTERISK);
+
+    private final String eTag;
+
+    /**
+     * Creates a new instance of {@link ETag}.
+     *
+     * @param eTag The HTTP entity tag string value.
+     */
+    public ETag(String eTag) {
+        checkValidETag(eTag);
+        this.eTag = eTag;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == this) {
+            return true;
+        }
+
+        if (!(o instanceof ETag)) {
+            return false;
+        }
+
+        ETag oETag = (ETag) o;
+
+        return this.eTag.equals(oETag.eTag);
```

Comment:
it seems null etag is considered valid, so this would throw along with has code. Should we allow null etags?

Reply:
Yes. Good point. I will update it to allow null ETag

Reply:
@lmolkova updated. Can you take another look? Thank you!